src/Ctype/Ctype.php

```diff
@@ -0,0 +1,246 @@
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_print($text)
+    {
+        return !empty($text) &&
+            !preg_match(
+                '/[^ -~]/',
+                self::convert_int_to_char_for_ctype($text)
+            );
+    }
+
+    /**
+     * Returns TRUE if every character in text is printable, but neither letter, digit or blank, FALSE otherwise.
+     *
+     * @see http://php.net/manual/en/function.ctype-punct.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_punct($text)
+    {
+        return !empty($text) &&
+            !preg_match(
+                '/[^!-\/\:-@\[-`\{-~]/',
+                self::convert_int_to_char_for_ctype($text)
+            );
+    }
+
+    /**
+     * Returns TRUE if every character in text creates some sort of white space, FALSE otherwise. Besides the blank character this also includes tab, vertical tab, line feed, carriage return and form feed characters.
+     *
+     * @see http://php.net/manual/en/function.ctype-space.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_space($text)
+    {
+        return !empty($text) &&
+            !preg_match(
+                '/[^\s]/',
+                self::convert_int_to_char_for_ctype($text)
+            );
+    }
+
+    /**
+     * Returns TRUE if every character in text is an uppercase letter.
+     *
+     * @see http://php.net/manual/en/function.ctype-upper.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     *
+     */
+    public static function ctype_upper($text)
+    {
+        return !empty($text) &&
+            !preg_match(
+                '/[^A-Z]/',
+                self::convert_int_to_char_for_ctype($text)
+            );
+    }
+
+    /**
+     * Returns TRUE if every character in text is a hexadecimal 'digit', that is a decimal digit or a character from [A-Fa-f] , FALSE otherwise.
+     *
+     * @see http://php.net/manual/en/function.ctype-xdigit.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_xdigit($text)
+    {
+        return !empty($text) &&
+            !preg_match(
+                '/[^A-Fa-f0-9]/',
+                self::convert_int_to_char_for_ctype($text)
+            );
+    }
+
+    /**
+     * Converts integers to their char versions according to normal ctype behaviour, if needed.
+     *
+     * @internal
+     *
+     * @param string|int $int
+     *
+     * @return string
+     */
+    private static function convert_int_to_char_for_ctype($int)
+    {
+        if(!is_integer($int)) {
+            return $int;
+        }
+
+        if ($int < -128 || $int > 255) {
+            return (string) $int;
+        }
+
+        if ($int < 0) {
+            $int += 256;
+        }
+
+        return chr($int);
```

Comment:
should have a backslash `\chr(...)`
(see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3048 for some background on this)